rest-service/manager_rest/blueprints_manager.py

```diff
@@ -503,14 +496,135 @@ def create_deployment(self, blueprint_id, deployment_id, inputs=None):
+        for raw_node in raw_nodes:
+            num_instances = raw_node['instances']['deploy']
+            nodes.append(models.DeploymentNode(
+                id=raw_node['name'],
+                deployment_id=deployment_id,
+                blueprint_id=blueprint_id,
+                type=raw_node['type'],
+                type_hierarchy=raw_node['type_hierarchy'],
+                number_of_instances=num_instances,
+                planned_number_of_instances=num_instances,
+                deploy_number_of_instances=num_instances,
+                host_id=raw_node['host_id'] if 'host_id' in raw_node else None,
+                properties=raw_node['properties'],
+                operations=raw_node['operations'],
+                plugins=raw_node['plugins'],
+                plugins_to_install=raw_node.get('plugins_to_install'),
+                relationships=self._prepare_node_relationships(raw_node)
+            ))
+        return nodes
+
+    def store_deployment_nodes(self, nodes):
+        for node in nodes:
+            self.sm.put_node(node)
+
+    @staticmethod
+    def prepare_deployment_node_instances_for_storage(deployment_id,
+                                                      dsl_node_instances,
+                                                      ):
+        node_instances = []
+        for node_instance in dsl_node_instances:
+            instance_id = node_instance['id']
+            node_id = node_instance['node_id']
+            relationships = node_instance.get('relationships', [])
+            host_id = node_instance.get('host_id')
+            instance = models.DeploymentNodeInstance(
+                id=instance_id,
+                node_id=node_id,
+                host_id=host_id,
+                relationships=relationships,
+                deployment_id=deployment_id,
+                state='uninitialized',
+                runtime_properties={},
+                version=None)
+            node_instances.append(instance)
+
+        return node_instances
+
+    def store_deployment_node_instances(self, node_instances):
+        for node_instance in node_instances:
+            self.sm.put_node_instance(node_instance)
+
+    def _create_deployment_nodes(self, blueprint_id, deployment_id, plan,
+                                 node_ids=None):
+        nodes = self.prepare_deployment_nodes_for_storage(
+            blueprint_id,
+            deployment_id,
+            plan,
+            node_ids)
+        self.store_deployment_nodes(nodes)
+
+    def _create_deployment_node_instances(self,
+                                          deployment_id,
+                                          dsl_node_instances):
+        node_instances = self.prepare_deployment_node_instances_for_storage(
+            deployment_id,
+            dsl_node_instances
+        )
+        self.store_deployment_node_instances(node_instances)
+
+    @staticmethod
+    def create_deployment_plan(blueprint_plan, inputs):
```

Comment:
refactor create_deployment_plan and prepare_deployment_plan.
look at how it was before.